### PR TITLE
Approaches for Atbash Cipher

### PR DESCRIPTION
**Pull request by safwansamsudeen**

### CODE REVIEW

File: exercises/practice/atbash-cipher/.approaches/mono-function/content.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ENCODING = {c: asc_low[i] for i, c in enumerate(asc_low[::-1])}
+
+def encode(text: str, decode: bool = False):
+    res = "".join(ENCODING.get(c, c) for c in text.lower() if c.isalnum())
+    return res if decode else " ".join(res[i:i+5] for i in range(0, len(res), 5))
+
+def decode(text: str):
+    return encode(text, True)
+```
+To explain the translation: we use a dict comprehension in which we reverse the ASCII lowercase digits, and enumerate through them - that is, `z` is 0, `y` is 1, and so on.
```
Comment (BethanyG) [inline exercises/practice/atbash-cipher/.approaches/mono-function/content.md:16]:
```suggestion
To explain the translation: we use a `dict` comprehension in which we reverse the ASCII lowercase digits, and enumerate through them - that is, `z` is 0, `y` is 1, and so on. 
```

**Review by BethanyG — CHANGES_REQUESTED** [review]
Hi @safwansamsudeen  - Thanks for submitting this!  Overall, it reads pretty well, but I am going to need to you change all the single letter variable names to something more readable in your code examples -- our track style is no single letter variable names - not even in comprehensions. Thanks for understanding.  🙂

**Comment by safwansamsudeen** [comment]
Hey sorry for the delay. Is `i` for index in loops allowed?

**Comment by BethanyG** [comment]
> Hey sorry for the delay. Is i for index in loops allowed?

Nope.  We are super-duper pendantic/conservative about it.  `index`, `item`, `member`, etc.

**Comment by BethanyG** [comment]
CI is broken at the moment.  Trying to get it fixed without having to regenerate all the test files again, but I might have to anyways.

**Comment by safwansamsudeen** [comment]
> Nope. We are super-duper pendantic/conservative about it.
As you wish ;).

I've also added a link to a Programiz article explaining `maketrans`, based on a student request.

**Comment by safwansamsudeen** [comment]
@BethanyG could we merge?

File: exercises/practice/atbash-cipher/.approaches/introduction.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+## Approach: separate functions
+We use `str.maketrans` to create the encoding. 
+In `encode`, we use a [generator expression][generator expression] in `str.join`.
+```python
+from string import ascii_lowercase
+ENCODING = str.maketrans(ascii_lowercase, ascii_lowercase[::-1])
+
+def encode(text: str):
+    res = "".join(chr for chr in text.lower() if chr.isalnum()).translate(ENCODING)
+    return " ".join(res[i:i+5] for i in range(0, len(res), 5))
```
Comment (BethanyG) [inline exercises/practice/atbash-cipher/.approaches/introduction.md:20]:
```suggestion
    return " ".join(res[index:index+5] for index in range(0, len(res), 5))
```

File: exercises/practice/atbash-cipher/.approaches/introduction.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+Notice that there the majority of the code is repetitive? 
+A fun way to solve this would be to keep it all inside the `encode` function, and merely chunk it if `decode` is False:
+For variation, this approach shows a different way to translate the text.
+```python
+from string import ascii_lowercase as asc_low
+ENCODING = {chr: asc_low[id] for id, chr in enumerate(asc_low[::-1])}
+
+def encode(text: str, decode: bool = False):
+    res = "".join(ENCODING.get(chr, chr) for chr in text.lower() if chr.isalnum())
+    return res if decode else " ".join(res[i:i+5] for i in range(0, len(res), 5))
```
Comment (BethanyG) [inline exercises/practice/atbash-cipher/.approaches/introduction.md:37]:
```suggestion
    return res if decode else " ".join(res[index:index+5] for index in range(0, len(res), 5))
```

**Review by BethanyG — CHANGES_REQUESTED** [review]
There's a couple of places where the variable names were single letters.  Beyond that, looks good.

**Pull request merged**